### PR TITLE
Fix parsing

### LINKED ISSUE #326: Prometheus still records invalid URL escapes
... which inflates the metrics.

I'm not sure where to fix this. We could have a static error in the `bittorrent` package that is used when URL query parsing fails. But that kind of hurts debugging.

### PR DESCRIPTION
**Pull request by mrd0ll4r**
Fixes #337 
Fixes #326 

Makes URL parsing about 15% faster on my machine:
```
BenchmarkParseQuery       	 2000000	      2016 ns/op	     387 B/op	       3 allocs/op
BenchmarkParseQuery       	 2000000	      2013 ns/op	     387 B/op	       3 allocs/op
BenchmarkParseQuerySimple 	 3000000	      1707 ns/op	     387 B/op	       3 allocs/op
BenchmarkParseQuerySimple 	 3000000	      1703 ns/op	     387 B/op	       3 allocs/op
BenchmarkURLParseQuery    	 2000000	      1852 ns/op	     463 B/op	       7 allocs/op
BenchmarkURLParseQuery    	 2000000	      1847 ns/op	     463 B/op	       7 allocs/op
```

`ParseQuery` is what we had before, `ParseQuerySimple` is this (a simplification of `url.parseQuery`), `URLParseQuery` is `url.ParseQuery`.

### CODE REVIEW

File: bittorrent/params_test.go [hunk 1]
```diff
@@ -27,6 +27,12 @@ var (
 	InvalidQueries = []string{
 		"/announce?" + "info_hash=%0%a",
 	}
+
+	// See https://github.com/chihaya/chihaya/issues/334.
+	panicQueries = []string{
```
Comment (jzelinskie) [inline bittorrent/params_test.go]: This seems to imply that these queries _should_ panic

File: bittorrent/params.go [hunk 2]
```diff
@@ -88,69 +94,55 @@ func ParseURLData(urlData string) (*QueryParams, error) {
...
+		if i := strings.Index(key, "="); i >= 0 {
+			key, value = key[:i], key[i+1:]
+		}
+		key, err = url.QueryUnescape(key)
+		if err != nil {
+			// QueryUnescape returns an error like "invalid escape: '%x'".
+			// But frontends record these errors to prometheus, which generates
+			// a lot of time series.
+			// We log it here for debugging instead.
+			log.Debugln(err)
```
Comment (jzelinskie) [inline bittorrent/params.go]: can we make this fit the newer log format?

```go
log.WithFields(log.Fields{"error": err}).Debug("failed to unescape query param")
```

Comment (mrd0ll4r) [inline bittorrent/params.go]: But that's gonna degrade performance even with debug logging disabled.

Comment (jzelinskie) [inline bittorrent/params.go]: ok, i guess we'll save that for a logging package PR

**Pull request merged**